actors/test/actor-tests/src/test/java/cloud/orbit/actors/test/ShutdownTest.java

```diff
@@ -122,5 +147,40 @@ public void asyncShutdownTest() throws ExecutionException, InterruptedException
 
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void invocationAfterShutdownTest()
+    {
+        final Stage stage = createStage();
+        fakeSync.semaphore("shouldDeactivate0").release();
+
+        final Deactivate deactivate = Actor.getReference(Deactivate.class, "0");
+        deactivate.doSomething();
+        stage.stop().join();
+        deactivate.doSomething().join();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void invocationDuringShutdownTest()
+    {
+        final Stage stage = createStage();
+
+        final Deactivate deactivate1 = Actor.getReference(Deactivate.class, "1");
+        final Deactivate deactivate2 = Actor.getReference(Deactivate.class, "2");
+
+        deactivate1.doSomething().join();
+        deactivate2.doSomething().join();
+
+        // Allow 1 to deactivate, but not 2
+        fakeSync.semaphore("shouldDeactivate1").release();
+
+        stage.stop();
```

Comment:
Although this passes, it changes what I was testing.  Your changes test what happens to a deactivated actor when you make an invocation **after** the stage has stopped. This throws an exception as it should.

My test sees what happens to a deactivated actor when you make an invocation **during** the shutdown of the stage.  This doesn't work, it re-activates the actor and doesn't throw an exception.